connect.go

```diff
@@ -288,9 +288,13 @@ type Spec struct {
 // On both the client and the server, Protocol is the RPC protocol in use.
 // Currently, it's either [ProtocolConnect], [ProtocolGRPC], or
 // [ProtocolGRPCWeb], but additional protocols may be added in the future.
+//
+// Query contains the query parameters for the request. For the server, this
+// will reflect the actual query parameters sent. For the client, it is unset.
 type Peer struct {
 	Addr     string
 	Protocol string
+	Query    url.Values
```

Comment:
Let's also add a `// server-only` to this field.

Reply:
FYI: I followed this onto the next PR, since I did merge this back into get-support preemptively. Sorry!